### ISSUE TITLE
authenticateEmail with username fails

### ISSUE THREAD

**Issue by nathanjde** [heroiclabs/nakama-js#87]
[The docs on email authentication](https://heroiclabs.com/docs/authentication/#email) explain that you can specify a username when creating a new account when calling `authenticateEmail`, e.g. `const session = await client.authenticateEmail(email, password, create, "mycustomusername");`

Full snippet that reproduces the issue with a local docker setup of Nakama using the [Docker Compose file here](https://heroiclabs.com/docs/install-docker-quickstart/#running-nakama-with-docker-compose) with default settings:

```
const client = new nakamajs.Client('defaultkey', '127.0.0.1', '7350', false);
const socket = client.createSocket(false, false);
client.authenticateEmail('abc@test.com', 'Password1', true, 'mycustomusername');
```

I can see the resulting request from this snippet in the Chrome dev tools network tab to `http://127.0.0.1:7350/v2/account/authenticate/email?` (no url params) with the body

```
{"email":"abc@test.com","password":"Password1","vars":true}
```

This request fails with status code 400 and no body. The curl example in the docs indicates the username needs to be sent as a URL parameter, but the javascript library doesn't appear to add it as a url parameter.

Using versions:
* Javascript library: 2.1.3 (iife)
* Nakama server: Docker Compose setup on local machine, version 2.12.0+650b506a deployed with no changes from the compose file [here](https://heroiclabs.com/docs/install-docker-quickstart/#running-nakama-with-docker-compose)

**Comment by lugehorsam** [comment]
Nice catch @nathanjde -- we'll update this and a couple other authentication calls as part of our patch release this week.

**Comment by lugehorsam** [comment]
This has been resolved in v2.1.4
